runtime/str.go

```diff
@@ -711,8 +755,6 @@ func adjustIndex(i, l int) int {
 		i = 0
 	case i < 0:
 		i += l
-	case i > l:
-		i = l
```

Comment:
It is usefull for adjusting negative values like [-1:], but not for bound checking I guess.
I am now adding this bound checkout before slicing.
```
       if start > end || start > l {
		return NewInt(-1).ToObject(), nil
	}
```